### PR TITLE
libsowc - remove legacy Errata.

### PR DESCRIPTION
**Pull request by SolidWallOfCode**
I had to make some non-trivial changes in the JSON-RPC, @brbzull0 needs to review and, I expect, ask for some changes.

### CODE REVIEW

**Comment by brbzull0** [comment]
[approve ci]

File: mgmt/rpc/jsonrpc/json/YAMLCodec.h [hunk 1]
```diff
@@ -174,41 +174,31 @@ class yamlcpp_json_encoder
...
+    json << YAML::Key << "message" << YAML::Value << errata.code().message();
+    if (!errata.is_ok()) {
       json << YAML::Key << "data";
       json << YAML::BeginSeq;
       for (auto const &err : errata) {
-        json << YAML::BeginMap;
-        json << YAML::Key << "code" << YAML::Value << err.getCode();
-        json << YAML::Key << "message" << YAML::Value << err.text();
-        json << YAML::EndMap;
+        json << YAML::Value << std::string(err.text()); // FIXME!! Shouldn't need a @c std::string
```
Comment (brbzull0) [inline mgmt/rpc/jsonrpc/json/YAMLCodec.h:195]: This is an important change, we need `code` and `message` as part of the aditional `data` message information.

**Review by brbzull0 — CHANGES_REQUESTED** [review]

File: mgmt/rpc/jsonrpc/json/YAMLCodec.h (hunk 1 above)
Comment (SolidWallOfCode) [reply mgmt/rpc/jsonrpc/json/YAMLCodec.h:195]: Hmmm. @bneradt made me take that out and overall I agree. We should also think about tracking the severity.

File: mgmt/config/FileManager.cc [hunk 2]
```diff
@@ -163,17 +161,20 @@ FileManager::getConfigObj(const char *fileName, ConfigManager **rbPtr)
...
 
   std::lock_guard<std::mutex> guard(_callbacksMutex);
   for (auto const &call : _configCallbacks) {
     if (auto const &r = call(fileName, configName); !r) {
       Debug("filemanager", "something back from callback %s", fileName.c_str());
-      std::for_each(r.begin(), r.end(), [&ret](auto &&e) { ret.push(e); });
+      if (ret.empty()) {
+        ret.note("Errors while reloading configurations.");
+      }
+      ret.note(r);
```
[the remark this answers is not in the document]
Comment (SolidWallOfCode) [reply mgmt/config/FileManager.cc:177]: I disagree. The check provides additional context for the more local error, e.g. whether it happened during configuration loading, or key file loading, etc.

File: mgmt/rpc/jsonrpc/json/YAMLCodec.h (hunk 1 above)
Comment (brbzull0) [reply mgmt/rpc/jsonrpc/json/YAMLCodec.h:195]: The [data](https://www.jsonrpc.org/specification#error_object) is user defined information for the specific higher error, we use this structure to show inner error of an operation, like :

```json
{
   "jsonrpc": "2.0",
   "id": "msg001",
   "error": {

      "code": 9,
      "message": "Error during execution",
      "data": [{
         "code": 3000,
         "message": "Server already draining."
         }]
   }
}
```
We can include the severity inside each inner error object.

```json
"data": [{
    "code": 3000,
    "severity": "Fatal"
    "message": "Server already draining."
}]
```

File: mgmt/config/FileManager.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (brbzull0) [reply mgmt/config/FileManager.cc:177]: I read those backwards. you are right, please ignore!

**Comment by bneradt** [comment]
[approve ci]

**Comment by brbzull0** [comment]
I will continue with this PR.

**Comment by brbzull0** [comment]
Work continues here -> https://github.com/apache/trafficserver/pull/10986

**Pull request closed without merging**